core/azure-core-amqp/src/main/java/com/azure/core/amqp/Retry.java

```diff
@@ -109,46 +91,35 @@ public int getMaxRetryCount() {
     /**
      * Calculates the amount of time to delay before the next retry attempt.
      *
-     * @param lastException The last exception that was observed for the operation to be retried.
+     * @param baseWaitTime The amount of time to base the suggested retry interval on. This should be used as
+     *         the minimum interval returned under normal circumstances.
      * @param remainingTime The amount of time remaining for the cumulative timeout across retry attempts.
-     * @return The amount of time to delay before retrying the associated operation; if {@code null},
-     * then the operation is no longer eligible to be retried.
+     * @return The amount of time to delay before retrying the associated operation; if {@code null}, then the operation
+     *         is no longer eligible to be retried.
      */
-    public Duration getNextRetryInterval(Exception lastException, Duration remainingTime) {
-        int baseWaitTime = 0;
-
-        if (!isRetriableException(lastException) || retryCount.get() >= maxRetryCount) {
-            return null;
-        }
-
-        if (!(lastException instanceof AmqpException)) {
+    public Duration getNextRetryInterval(Duration baseWaitTime, Duration remainingTime) {
```

Comment:
For helping custom retry policies, I think it's important to expose the exception that was last encountered so that there is full context on why an operation failed and an external policy can make decisions based on it.

Reply:
We moved this out because it felt like an additional responsibility for the policy to take care of and could be pushed to the caller. But in regards to AMQP, since transient errors are built into the protocol, I can see the value in adding back a parameter that takes an AmqpException.

@JonathanGiles may have some more insights to this. 🤔